source/server/guarddog_impl.cc

```diff
@@ -218,10 +218,10 @@ void GuardDogImpl::stopWatching(WatchDogSharedPtr wd) {
 void GuardDogImpl::start(Api::Api& api) {
   Thread::LockGuard guard(mutex_);
   // See comments in WorkerImpl::start for the naming convention.
+  loop_timer_->enableTimer(std::chrono::milliseconds(0));
```

Comment:
Why this order change?

Reply:
This enableTimer happens in a different thread than the dispatcher run, but it is fine if I move the call up before the dispatcher thread is started.  Another possibly more clear alternative would be to move this enable call to the thread lambda just before the call to dispatcher_->run.

Let me know if you would prefer to move it to the lambda.

Reply:
> Another possibly more clear alternative would be to move this enable call to the thread lambda just before the call to dispatcher_->run.

Yeah this seems better or a comment is fine. Seems like other CI issues will require another push anyway. Thank you!

/wait

Reply:
Moved the enable call and debug the issue with the test.